File: domain/src/main/java/home/oleg/placenearme/repositories/VenueRepository.java
package home.oleg.placenearme.repositories;

import java.util.List;

import home.oleg.placenearme.models.Section;
import home.oleg.placenearme.models.Venue;
import io.reactivex.Single;

public interface VenueRepository {

    Single<List<Venue>> getRecommendedBySection(Section.Type section, VenueRequestParams requestParams);

    Single<List<Venue>> search(String query, VenueRequestParams requestParams);

}

File: domain/src/main/java/home/oleg/placenearme/repositories/VenueRepository.kt
package home.oleg.placenearme.repositories

import home.oleg.placenearme.models.Section
import home.oleg.placenearme.models.Venue
import io.reactivex.Single

interface VenueRepository {

    fun getRecommendedBySection(section: Section, requestParams: VenueRequestParams): Single<List<Venue>>

    fun search(query: String, requestParams: VenueRequestParams): Single<List<Venue>>

}
